src/pages/my_page/user_profile/UserOrcid.tsx

```diff
@@ -61,29 +94,31 @@ export const UserOrcid = ({ user }: UserOrcidProps) => {
           method: 'PATCH',
           data: { orcid },
         });
-        if (isSuccessStatus(addOrcidResponse.status)) {
-          dispatch(setNotification({ message: t('feedback.success.update_orcid'), variant: 'success' }));
-          cristinPersonQuery.refetch();
-        } else if (isErrorStatus(addOrcidResponse.status)) {
-          dispatch(setNotification({ message: t('feedback.error.update_orcid'), variant: 'success' }));
+        const orcidCredentials = getOrcidCredentials(history.location.search, orcidInfoResponse.data.id);
+        if (!orcidCredentials) {
+          dispatch(setNotification({ message: t('feedback.error.storing_orcid_credentials'), variant: 'warning' }));
+        } else {
+          const postOrcidCredentialsResponse = await postOrcidCredentials(orcidCredentials);
+          if (isSuccessStatus(addOrcidResponse.status) && isSuccessStatus(postOrcidCredentialsResponse.status)) {
+            dispatch(setNotification({ message: t('feedback.success.update_orcid'), variant: 'success' }));
+          } else if (isErrorStatus(addOrcidResponse.status) || isErrorStatus(postOrcidCredentialsResponse.status)) {
+            dispatch(setNotification({ message: t('feedback.error.update_orcid'), variant: 'success' }));
+          }
         }
       }
-      history.push(UrlPathTemplate.MyPageMyProfile);
       setIsAddingOrcid(false);
+      history.replace(UrlPathTemplate.MyPageMyProfile);
```

Comment:
Denne burde kanksje endres til `UrlPathTemplate.MyPageMyPersonalia` ettersom det er gjort noen endringer med de nye menyene.
```suggestion
      history.replace(UrlPathTemplate.MyPageMyPersonalia);
```